### ISSUE TITLE
failed in building shared library and demo

### ISSUE THREAD

**Issue by zealerww** [leetal/ios-cmake#29]
when build shared example( add flag -DBUILD_SHARED=1), there is sth. wrong
following 
```
cd example/example-lib
mkdir build
cd build
cmake .. -DCMAKE_TOOLCHAIN_FILE=../../ios.toolchain.cmake -DIOS_PLATFORM=SIMULATOR64 -DBUILD_SHARED=1
make
```

```
Scanning dependencies of target example
[ 20%] Building CXX object CMakeFiles/example.dir/HelloWorld.cpp.o
[ 40%] Building CXX object CMakeFiles/example.dir/HelloWorldIOS.mm.o
[ 60%] Linking CXX shared library libexample.dylib
[ 60%] Built target example
Scanning dependencies of target helloworld
[ 80%] Building CXX object CMakeFiles/helloworld.dir/main.cpp.o
[100%] Linking CXX executable helloworld
Undefined symbols for architecture x86_64:
  "HelloWorld::helloWorld()", referenced from:
      _main in main.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [helloworld] Error 1
make[1]: *** [CMakeFiles/helloworld.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by leetal** [comment]
I'll take a look at this today. What version of cmake do you use?

**Comment by zealerww** [comment]
cmake verison 3.12.4

cmake info
```
-- Building with Xcode version: 10.2
-- Configuring iOS build for platform: SIMULATOR64, architecture(s): x86_64
-- Using manually set SDK path: /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator12.2.sdk for platform: SIMULATOR64
-- Using the default min-version since IOS_DEPLOYMENT_TARGET not provided!
-- Disabling bitcode support by default on simulators. ENABLE_BITCODE not provided for override!
-- Enabling ARC support by default. ENABLE_ARC not provided!
-- Hiding symbols visibility by default. ENABLE_VISIBILITY not provided!
-- Using C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang
-- Using CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++
-- Using libtool: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool
-- Using a data_ptr size of 8
-- Building for minimum iOS version: 8.0 (SDK version: 12.2)
-- Version flags set to: -mios-simulator-version-min=8.0
-- Disabling bitcode support.
-- Enabling ARC support.
-- Hiding symbols (-fvisibility=hidden).
-- Building with Xcode version: 10.2
-- Configuring iOS build for platform: SIMULATOR64, architecture(s): x86_64
-- Using manually set SDK path: /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator12.2.sdk for platform: SIMULATOR64
-- Disabling bitcode support by default on simulators. ENABLE_BITCODE not provided for override!
-- Using libtool: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool
-- Using a data_ptr size of 8
-- Building for minimum iOS version: 8.0 (SDK version: 12.2)
-- Version flags set to: -mios-simulator-version-min=8.0
-- Disabling bitcode support.
-- Enabling ARC support.
-- Hiding symbols (-fvisibility=hidden).
-- The C compiler identification is AppleClang 10.0.1.10010046
-- The CXX compiler identification is AppleClang 10.0.1.10010046
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Building shared version...
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/wuwei/ios-cmake/example/example-lib/build
```

**Comment by leetal** [comment]
Fixed in [v3.0.0](https://github.com/leetal/ios-cmake/tree/3.0.0) 🚀 This was due to the fact that the "Application" being built and linked was not actually working in the first place and when the symbols was hidden the builds also failed.
